sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -752,9 +752,30 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
       expressions.exists(_.exists(_.semanticEquals(expr)))
     }
 
+    def checkOuterReferenceUtil(attrs: Seq[Attribute], outer: OuterReference): Unit = {
```

Comment:
This is only called once, can we inline it?

Reply:
sure done.